File: apps/plant_shop_java/Structure_en_application_distribuée/plant_shop_java_quarkus_distribuée/order-service/src/models/OrderItem.java
package models;

import java.math.BigDecimal;

public final class OrderItem {
    public int id;
    public int orderId;
    public int plantId;
    public int quantity;
    public BigDecimal price;

    public OrderItem(int id, int orderId, int plantId, int quantity, BigDecimal price) {
        this.id = id;
        this.orderId = orderId;
        this.plantId = plantId;
        this.quantity = quantity;
        this.price = price;
    }

    public OrderItem(int orderId, int plantId, int quantity, BigDecimal price) {
        this(0, orderId, plantId, quantity, price);
    }

    // Constructeur par défaut nécessaire pour la désérialisation JSON
    public OrderItem() {}
}

File: apps/plant_shop_c-sharp/plant_shop_asp_EF_core/Models/OrderItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace plant_shop_asp_EF_core.Models
{
    /// <summary>
    /// Ligne de commande reliant une commande a une plante.
    /// </summary>
    [Table("order_items")]
    public class OrderItem
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("order_id")]
        public int OrderId { get; set; }

        [Required]
        [Column("plant_id")]
        public int PlantId { get; set; }

        [Required]
        [Column("quantity")]
        public int Quantity { get; set; }

        [Required]
        [Column("price", TypeName = "decimal(18, 2)")]
        public decimal Price { get; set; } // Prix au moment de la commande

        // Relations
        [ForeignKey("PlantId")]
        public Plant? Plant { get; set; }

        [JsonIgnore] // Evite la référence circulaire
        [ForeignKey("OrderId")]
        public Order? Order { get; set; }
    }
}
